tasks/encryption-at-rest.yml

```diff
@@ -56,49 +7,79 @@
+    dest: "{{ mariadb_file_key_management_dir }}/{{ item.custom_name }}"
     owner: mysql
     group: mysql
     mode: 0600
-    backup: true
-  no_log: "{{ not mariadb_debug }}"
-  register: _mariadb_encryption_password_copy
-  when: mariadb_file_key_management_override or not _mariadb_filekeypass_stat.stat.exists
+  loop:
+    - default_name: keyfile
+      custom_name: keyfile
+    - default_name: keyfile.key
+      custom_name: "{{ mariadb_file_key_management_filekey }}"
+  when:
+    - not _mariadb_filekeypass_stat.stat.exists or mariadb_file_key_management_override
+    - ( mariadb_encryption_keys is defined ) and ( mariadb_encryption_keys | length > 0 )
+    - ( mariadb_encryption_password is defined ) and ( mariadb_encryption_password | length > 0 )
 
-# keyfile.enc
-- name: "Setup the encrypted keyfile"
+- name: "Create encrypted file automatically"
+  when:
+    - not _mariadb_filekeypass_stat.stat.exists or mariadb_file_key_management_override
+    - mariadb_encryption_keygen_auto
   block:
-    - name: "Generate the encrypted key file using openssl enc"
-      ansible.builtin.command: "openssl enc -aes-256-cbc -md sha1 -pass file:{{ mariadb_file_key_management_filekey }} -out {{ mariadb_file_key_management_filename }}"
-      args:
+    - name: "Generate key for encryption"
+      ansible.builtin.shell:
+        cmd: "openssl rand -hex 128 | tee {{ mariadb_file_key_management_filekey }}"
         chdir: "{{ mariadb_file_key_management_dir }}"
-        stdin: "{{ mariadb_encryption_keys_string }}"
-      no_log: "{{ not mariadb_debug }}"
-      changed_when: true
-      when: mariadb_file_key_management_override or
-            not _mariadb_fileenc_stat.stat.exists or
-            _mariadb_keyssha1 is changed or
-            _mariadb_encryption_password_copy is changed
+      register: _mariadb_key_generation
+      changed_when: _mariadb_key_generation.rc != 0
+
+    - name: "Create a variable containing the complete opennssl command to generate the correct number of encryption keys"
+      ansible.builtin.set_fact:
+        _mariadb_openssl_genkey_cmd: "{{ _mariadb_openssl_genkey_cmd | d('') + 'openssl rand -hex 32 ;' }}"
+      loop: "{{ range(1, mariadb_encryption_keygen_count | int + 1) | list }}"
 
-    - name: "Set owner and group on encryted key file"
-      ansible.builtin.file:
-        path: "{{ _mariadb_file_key_management_filename_path }}"
+    - name: "Generate the encryption keys with openssl"
+      ansible.builtin.shell: "{{ _mariadb_openssl_genkey_cmd }}"
+      register: _mariadb_opensslcmd_encryption_keys
+      changed_when: _mariadb_opensslcmd_encryption_keys.rc != 0
+      tags:
+        - skip_ansible_lint
+
+    - name: Build the encryption keys string version
+      ansible.builtin.set_fact:
+        mariadb_encryption_keys_string: |
+          {% for key in _mariadb_opensslcmd_encryption_keys.stdout_lines %}
+          {{ loop.index }};{{ key }}
+          {% endfor %}
+
+    - name: "Generate keyfile"
```

Comment:
Rename this task to reflect the action of simply copying the appropriate content and not actually **generating** that content.
Maybe use a name along the lines of **Copy encryption keys inside the file keyfile** or **Store encryption keys locally**